.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,198 @@
+name: Build, test and deploy
```

Comment:
have you thought about having these as separate jobs rather than one single job ?

Reply:
The idea with a single workflow was to test and then deploy a fixed target. The graph view of the jobs that the workflow contains that Github produces is quite nice and descriptive too, and captures the parallel parts well. The build process isn't completely safe here - there are clear interactions with docker caching because of syncing latest packages, and building from vcs - so keeping the parts together seemed advantageous.

Reply:
I think we should rename the file `build-test.yaml` as that is what the pipeline should do, the docker publishing is just a nice-to-have ;-)